builtin/providers/docker/resource_docker_container_funcs.go

```diff
@@ -112,6 +117,46 @@ func resourceDockerContainerCreate(d *schema.ResourceData, meta interface{}) err
 		hostConfig.Links = stringSetToStringSlice(v.(*schema.Set))
 	}
 
+	if v, ok := d.GetOk("memory"); ok {
+		memory := int64(v.(int))
+		if memory > 0 {
```

Comment:
Instead of this check, we should be able to trust that `d.GetOk("memory")` will return a non-zero value, and instead put in a check for negative values into the `ValidateFunc` for this schema such that obviously bad values are raised to the user.